rust/blockstore/src/types.rs

```diff
@@ -277,19 +278,44 @@ impl<
         }
     }
 
-    pub async fn get_range<'prefix, PrefixRange, KeyRange>(
+    pub fn get_range_stream<'prefix, PrefixRange, KeyRange>(
         &'referred_data self,
         prefix_range: PrefixRange,
         key_range: KeyRange,
-    ) -> Result<Vec<(K, V)>, Box<dyn ChromaError>>
+    ) -> impl Stream<Item = Result<(K, V), Box<dyn ChromaError>>> + 'referred_data + Send
     where
-        PrefixRange: RangeBounds<&'prefix str> + Clone,
-        KeyRange: RangeBounds<K> + Clone,
+        PrefixRange: RangeBounds<&'prefix str> + Clone + Send + 'referred_data,
+        KeyRange: RangeBounds<K> + Clone + Send + 'referred_data,
+        K: Sync + Send,
```

Comment:
same question, why does it have to be Sync + Send?

Reply:
that's the only concern I have with this PR - restricting our key and value type to be Send and Sync might be something that might bite us back in future